oshdb-api/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestOSHDBFilter.java

```diff
@@ -79,12 +93,26 @@ public void testAggregateFilter() throws Exception {
 
   @Test
   public void testAggregateFilterObject() throws Exception {
-    MapReducer<OSMEntitySnapshot> mr = createMapReducer();
-    SortedMap<OSMType, Integer> result = mr
+    SortedMap<OSMType, Integer> result = createMapReducerOSMEntitySnapshot()
         .aggregateBy(x -> x.getEntity().getType())
         .filter(filterParser.parse("(geometry:polygon or geometry:other) and building=*"))
         .count();
 
     assertEquals(42, result.get(OSMType.WAY).intValue());
   }
+
+  @Test
+  public void testFilterGroupByEntity() throws Exception {
```

Comment:
Could be split in two separate tests. If it would contain more than the two asserts, SonarLint would recommend that, but I think it's okay for now.

Reply:
sure, but I think this is good enough for now ;)